src/client/automation/get-element.js

```diff
@@ -73,8 +69,33 @@ export function fromPoint (x, y, expectedElement) {
     var linkRect = expectedElement.getBoundingClientRect();
 
     return findElementOrNonEmptyChildFromPoint(linkRect.right - 1, linkRect.top + 1, expectedElement) ||
-           findElementOrNonEmptyChildFromPoint(linkRect.left + 1, linkRect.bottom - 1, expectedElement) ||
-           topElement;
+                 findElementOrNonEmptyChildFromPoint(linkRect.left + 1, linkRect.bottom - 1, expectedElement) ||
+                 topElement;
+}
+
+export function fromPoint (x, y, expectedElement) {
+    var expectedElementDefined = expectedElement && domUtils.isDomElement(expectedElement);
+
+    return getElementFromPoint(x, y)
+        .then(topElement => {
+            // NOTE: when trying to get an element by elementFromPoint in iframe and the target
+            // element is under any of shadow-ui elements, you will get null (only in IE).
+            // In this case, you should hide a top window's shadow-ui root to obtain an element.
+            if (!topElement) {
+                return getElementFromPoint(x, y, true)
+                    .then(element => {
+                        if (!expectedElementDefined || !element || element === expectedElement)
```

Comment:
maybe it's better to move this conditions to `correctTopElementByExpectedElement` too, since this is part of corrections logic?